data-prepper-plugins/otlp-source/src/main/java/org/opensearch/dataprepper/plugins/source/otlp/OTLPSource.java

```diff
@@ -0,0 +1,180 @@
+          (int) (otlpSourceConfig.getRequestTimeoutInMillis() * 0.8),
+          otlpSourceConfig.getTracesOutputFormat() == OTelOutputFormat.OPENSEARCH ? new OTelProtoOpensearchCodec.OTelProtoDecoder() : new OTelProtoStandardCodec.OTelProtoDecoder(),
+          buffer, pluginMetrics);
+
+      ServerConfiguration serverConfiguration = ConvertConfiguration.convertConfiguration(otlpSourceConfig);
+      CreateServer createServer = new CreateServer(serverConfiguration, LOG, pluginMetrics, "otlp", pipelineName);
+      CertificateProvider certificateProvider = null;
+      if (otlpSourceConfig.isSsl() || otlpSourceConfig.useAcmCertForSSL()) {
+        certificateProvider = certificateProviderFactory.getCertificateProvider();
+      }
+
+      List<GRPCServiceConfig<?, ?>> serviceConfigs = new ArrayList<>();
+      serviceConfigs.add(new GRPCServiceConfig<>(oTelLogsGrpcService, otlpSourceConfig.getLogsPath(),
+          LogsServiceGrpc.getExportMethod()));
+      serviceConfigs.add(new GRPCServiceConfig<>(oTelMetricsGrpcService, otlpSourceConfig.getMetricsPath(),
+          MetricsServiceGrpc.getExportMethod()));
+      serviceConfigs.add(new GRPCServiceConfig<>(oTelTraceGrpcService, otlpSourceConfig.getTracesPath(),
+          TraceServiceGrpc.getExportMethod()));
+
+      server = createServer.createGRPCServer(authenticationProvider, serviceConfigs, certificateProvider);
+
+      pluginMetrics.gauge(SERVER_CONNECTIONS, server, Server::numConnections);
+    }
+
+    try {
+      server.start().get();
+    } catch (ExecutionException ex) {
+      if (ex.getCause() != null && ex.getCause() instanceof RuntimeException) {
+        throw (RuntimeException) ex.getCause();
+      } else {
+        throw new RuntimeException(ex);
+      }
+    } catch (InterruptedException ex) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(ex);
+    }
+    LOG.info("Started otlp_source on port " + otlpSourceConfig.getPort() + "...");
+  }
+
+  @Override
+  public void stop() {
+    if (server != null) {
+      try {
+        server.stop().get();
+      } catch (ExecutionException ex) {
+        if (ex.getCause() != null && ex.getCause() instanceof RuntimeException) {
+          throw (RuntimeException) ex.getCause();
+        } else {
+          throw new RuntimeException(ex);
+        }
+      } catch (InterruptedException ex) {
+        Thread.currentThread().interrupt();
+        throw new RuntimeException(ex);
+      }
+    }
+    LOG.info("Stopped otlp_source.");
+  }
+
+  private GrpcAuthenticationProvider createAuthenticationProvider(final PluginFactory pluginFactory) {
+    final PluginModel authenticationConfiguration = otlpSourceConfig.getAuthentication();
+
+    if (authenticationConfiguration == null || authenticationConfiguration.getPluginName()
+        .equals(GrpcAuthenticationProvider.UNAUTHENTICATED_PLUGIN_NAME)) {
+      LOG.warn("Creating otlp-source without authentication. This is not secure.");
```

Comment:
```suggestion
      LOG.warn("Creating otlp source without authentication. This is not secure.");
```

No hyphen is needed here.